blockchain/blockchain.go

```diff
@@ -33,7 +33,11 @@ type Reader interface {
 
 	TransactionByHash(hash *felt.Felt) (transaction core.Transaction, err error)
 	TransactionByBlockNumberAndIndex(blockNumber, index uint64) (transaction core.Transaction, err error)
+	TxBlockNumberAndIndexByHash(hash *felt.Felt) (blockNumber uint64, index uint64, err error)
```

Comment:
nit: Because it's a new function, let's try using the new types.
```suggestion
	BlockNumberAndIndexByTxHash(hash *felt.TransactionHash) (blockNumber uint64, index uint64, err error)
```